vavr/src/test/java/io/vavr/MatchTest.java

```diff
@@ -315,17 +315,12 @@ public void shouldDecomposeListOfTuple3() {
         assertThat(actual).isEqualTo("(begin, 10, 4.5)::List((middle, 11, 0.0), (end, 12, 1.2))");
     }
 
-    @SuppressWarnings("UnnecessaryLocalVariable")
     @Test
     public void shouldDecomposeListWithNonEmptyTail() {
         final List<Option<Number>> intOptionList = List.of(Option.some(1), Option.some(2.0));
         final String actual = Match(intOptionList).of(
-                Case($Cons($Some($(1)), $Cons($Some($(2.0)), $())), (x, xs) -> {
-                    // types are inferred correctly!
-                    final Some<Number> head = x;
-                    final List<Option<Number>> tail = xs;
-                    return head + "::" + tail;
```

Comment:
I would keep the block. It was intended to make the types explicitly visible (even if vars are not used).
The difference to `(Some<Number> x, List<Option<Number>> xs) -> x + "::" + xs)` is that we have `List<Option<Number>> tail = xs`. I.e. we show which general type is inferred by Match.

Reply:
Yeah, except that it's not working in `Java 9`, different types are inferred for some reason.
If I specify it explicitly, as done here, it's working.
Not sure how to fix it otherwise.

Reply:
Fixed, thanks!